Source/Scene/BatchTable.js

```diff
@@ -0,0 +1,475 @@
+        var width = numberOfAttributes * instancesPerWidth;
+        var height = Math.ceil(numberOfInstances / instancesPerWidth);
+
+        var stepX = 1.0 / width;
+        var centerX = stepX * 0.5;
+        var stepY = 1.0 / height;
+        var centerY = stepY * 0.5;
+
+        this._textureDimensions = new Cartesian2(width, height);
+        this._textureStep = new Cartesian4(stepX, centerX, stepY, centerY);
+        this._pixelDatatype = pixelDatatype;
+        this._texture = undefined;
+
+        var batchLength = width * height * 4;
+        this._batchValues = pixelDatatype === PixelDatatype.FLOAT ? new Float32Array(batchLength) : new Uint8Array(batchLength);
+        this._batchValuesDirty = false;
+    }
+
+    defineProperties(BatchTable.prototype, {
+        /**
+         * The attribute descriptions.
+         * @memberOf BatchTable.prototype
+         * @type {Object[]}
+         * @readonly
+         */
+        attributes : {
+            get : function() {
+                return this._attributes;
+            }
+        },
+        /**
+         * The number of instances.
+         * @memberOf BatchTable.prototype
+         * @type {Number}
+         * @readonly
+         */
+        numberOfInstances : {
+            get : function () {
+                return this._numberOfInstances;
+            }
+        }
+    });
+
+    function getDatatype(attributes) {
+        var foundFloatDatatype = false;
+        var length = attributes.length;
+        for (var i = 0; i < length; ++i) {
+            if (attributes[i].componentDatatype !== ComponentDatatype.UNSIGNED_BYTE) {
+                foundFloatDatatype = true;
+            }
+        }
+        return foundFloatDatatype ? PixelDatatype.FLOAT : PixelDatatype.UNSIGNED_BYTE;
+    }
+
+    function getEntryType(attributes, attributeIndex) {
+        var componentsPerAttribute = attributes[attributeIndex].componentsPerAttribute;
+        if (componentsPerAttribute === 2) {
+            return Cartesian2;
+        } else if (componentsPerAttribute === 3) {
+            return Cartesian3;
+        } else if (componentsPerAttribute === 4) {
+            return Cartesian4;
+        }
+        return Number;
+    }
+
+    var scratchgetEntryCartesian4 = new Cartesian4();
+
+    /**
+     * Gets the value of an entry in the table.
+     *
+     * @param {Number} instanceIndex The index of the instance.
+     * @param {Number} attributeIndex The index of the attribute.
+     * @param {undefined|Cartesian2|Cartesian3|Cartesian4} [result] The object onto which to store the result. The type is dependent on the attribute's number of components.
+     * @returns {Number|Cartesian2|Cartesian3|Cartesian4} The attribute value stored for the instance.
+     *
+     * @exception {DeveloperError} instanceIndex is out of range.
+     * @exception {DeveloperError} attributeIndex is out of range.
+     */
+    BatchTable.prototype.getEntry = function(instanceIndex, attributeIndex, result) {
```

Comment:
Throughout is `Entry` the best term?  Seems like we just introduced it.  What is this called in Cesium primitives, an attribute, right?  Perhaps `getBatchedAttribute` and friends?